Front improvements

existe-deja opened the pull request:
Merge Arthur and Simon's code

achauffaille committed:
Smooth scroll + assets clean + Bouton scroll intro + css

Simonguerot committed:
Fix analytics + RS

achauffaille committed:
Menu style change scroll JS

achauffaille committed:
Bug btn scroll intro + ajustement scroll js dynamique

achauffaille committed:
Ancres + refresh auto resize + responsive

Ancres dynamiques pour optimiser le référencement

achauffaille committed:
Transitions menu fixed + js

achauffaille committed:
Animations CSS + revealscroll JS

achauffaille committed:
Ajustements css

Centrer élément earth horizontalement
…

achauffaille committed:
Logos Jaab SVG

Assets - logo jaab HD x3
Modification CSS - width logo
Modification JS - appel svg
Modification html - appel svg

achauffaille committed:
Optim animation + remove google social

achauffaille committed:
Watch demo video link dynamic + responsive

Le bouton sur la strate d’intro sur le mobile « Watch demo video » est
maintenant un vrai lien.
Son comportement est également géré en responsive avec les media-queries

achauffaille committed:
Centrer logo + pop-in video test

existe-deja commented on ui/css/style.css:
```diff
@@ -201,22 +210,86 @@ input[type='submit']{
   position: absolute;
   right: 14%;
 }
+.plain:nth-of-type(1) .pure-u-1:nth-of-type(2) a {
```
selecteur trop barbare, rajouter une classe sur l'élément concerné.

achauffaille replied:
Ok réglé

existe-deja commented on ui/css/style.css:
```diff
@@ -201,22 +210,86 @@ input[type='submit']{
+.plain:nth-of-type(1) .wrapper img {
+  
+}  
+.plain:nth-of-type(1) .pure-u-1:nth-of-type(2) .pure-pos {
```
pure-pos signifie quoi ?

achauffaille replied:
J'ai changé en pos-phone, ça correspond au placement du phone pour que le lien en dynamique dedans ai une position qui réagisse en fonction de la position de la div. Pas géré avant

existe-deja commented on ui/css/style.css:
```diff
@@ -201,22 +210,86 @@ input[type='submit']{
+  position: absolute;
+  color:#ffffff;
+}
+.plain:nth-of-type(1) .pure-u-1:nth-of-type(2) span:nth-of-type(1) {
```
selecteur trop barbare, rajouter une classe sur l'element concerné.

achauffaille replied:
Ok réglé

existe-deja commented on ui/css/style.css:
```diff
@@ -201,22 +210,86 @@ input[type='submit']{
+  -webkit-border-radius: 200px;
+  border-radius: 200px;
+}
+.plain:nth-of-type(1) .pure-u-1:nth-of-type(2) span:nth-of-type(2) {
```
selecteur trop barbare, rajouter une classe sur l'element concerné.

achauffaille replied:
Ok réglé

existe-deja commented on ui/css/style.css:
```diff
@@ -201,22 +210,86 @@ input[type='submit']{
+  position: relative;
+  top: 20px;,
+}
+.plain:nth-of-type(1) .pure-u-1:nth-of-type(3) a{
```
selecteur trop barbare, rajouter une classe sur l'element concerné.

achauffaille replied:
Ok réglé

Lacroute commented on ui/css/style.css:
```diff
@@ -431,14 +519,34 @@ footer .swag-si-tu-lis-ca li:last-child{
   .plain header p {
     width: auto;
   }
+  .plain:nth-of-type(1) p{
```
à ranger dans la section 1

Lacroute commented on ui/css/style.css:
```diff
@@ -431,14 +519,34 @@ footer .swag-si-tu-lis-ca li:last-child{
+  .plain:nth-of-type(1) p{
+    text-align:center;
+  }
+  .plain:nth-of-type(1) h1{
```
à ranger dans la section 1

existe-deja commented on ui/css/style.css:
```diff
@@ -448,6 +556,10 @@ footer .swag-si-tu-lis-ca li:last-child{
   }
+  .pure-menu img {
+    left: 50%;
+    margin-left: -90px;
```
-90px ?

achauffaille replied:
J'ai vu que le centrage de certains éléments peut se faire comme ça. ça correspond à la moitié de la taille de l'élément. Comme ça le centrage est parfait et responsive

existe-deja replied:
Oui oui sans soucis, je pensais que le logo faisait plus que 180px comme tu l'avais transformé en svg. Good pour oim.

existe-deja commented on ui/css/style.css:
```diff
@@ -461,6 +573,15 @@ footer .swag-si-tu-lis-ca li:last-child{
+  .plain:nth-of-type(1) .pure-u-1:nth-of-type(2) .pure-pos {
+    position:initial;
+  }
+  .plain:nth-of-type(1) .pure-u-1:nth-of-type(2) a {
```
Sélecteur trop complexe

achauffaille replied:
Ok réglé

existe-deja commented on ui/css/style.css:
```diff
@@ -488,8 +609,23 @@ footer .swag-si-tu-lis-ca li:last-child{
+
+/*  TEST ANIMATION  */
+
+#popPlayer {
```
?

achauffaille replied:
Ça c'était un test en cours que je vais poursuivre ;)

existe-deja replied:
du coup pour les prochains push, pousse que ce qui est fontionnel et prêt pour la prod comme ça la pull request passe plus vite :)

existe-deja commented on ui/home.htm:
```diff
@@ -1,66 +1,78 @@
       <ul class="pure-menu-list">
-          <li class="pure-menu-item"><a href="#" class="pure-menu-link">Presse</a></li>
-          <li class="pure-menu-item"><a href="#beta-testeur" class="pure-menu-link">Beta testeur</a></li>
+          <li class="pure-menu-item"><a href="#" id="press" class="pure-menu-link">Presse</a></li>
```
intérêt d'ajouter un id ?

achauffaille replied:
Je l'ai modifié en ID, cela te convient ? C'est parce que je l'utilise dans mon JS pour le cibler et c'était plus simple comme ça qu'avec des child et tout

existe-deja commented on ui/home.htm:
```diff
@@ -1,66 +1,78 @@
 
 <!-- Second section -->
-<section class="plain">
+<section class="plain" id="scroll1" class="plain">
```
changer le nom de l'id, genre intro ou first-pane ou whatever, mais pas de chiffre.
Double attribut class.

achauffaille replied:
Ok réglé
Ok réglé

existe-deja commented on ui/home.htm:
```diff
@@ -68,52 +80,52 @@
 
 <!-- Fourth section -->
-<section class="plain">
+<section id="scroll3" class="plain">
```
idem, du coup plutot partir sur third-panel et first panel. A quoi servent ces ids ?

achauffaille replied:
Ces id servent à récupérer l'élément en JS et à détecter leur position pour changer la couleur du menu

achauffaille replied:
Ok réglé

existe-deja commented on ui/home.htm:
```diff
@@ -121,40 +133,39 @@
 
 <!-- Sixth section -->
-<section class="plain">
+<section data-sr="over 1s, enter bottom, move 100px, scale 0%" id="scroll4" class="plain">
```
idem id.

achauffaille replied:
Ok réglé

existe-deja commented on ui/images/logo2.svg:
```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Renommer le ficher sans chiffre plutot couleur (lisibilité)

achauffaille replied:
No comprendo. Je n'ai pas touché à ça il me semble.

achauffaille replied:
Ok réglé
logo-white
logo-blue
logo-green

existe-deja commented on ui/images/logo3.svg:
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Renommer le ficher sans chiffre.

achauffaille replied:
Ok réglé

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
\ No newline at end of file
+});
+
+$(document).ready(function() {
```
ne sert à rien car déjà déclaré L01, remonter l'ensemble dans le premier callback.

achauffaille replied:
C'est réglé

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
+
+
+/* FUNCTION REFRESH */
+/*$(document).ready(function(){      
```
enlever les fonctions en commentaire

achauffaille replied:
Je l'ai laissée en commentaire pour qu'un vrai dev regarde comment on peut régler ça.
C'est pour refresh la position des éléments quand on redimensionne la page.
Car les calculs en JS ne sont pas mis à jour sinon et il y a donc des décalages.

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
+
+$(document).ready(function() {
+    $('.scrollTo').click( function() { // Au clic sur un élément
+      var page = $(this).attr('href'); // Page cible
```
préferer une déclaration de ce type :
var page = ...
  , speed = ...
  ;

achauffaille replied:
No comprendo.
Tu penses vraiment qu'ils vont aller aussi loin en une soutenance de 10min ?

achauffaille replied:
Il me semble que c'est réglé si j'ai bien compris

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
+});*/
+/* END FUNCTION REFRESH */
+            
+            var scroll_pos = 0;
```
idem pour la déclaration des fonctions.
Convention indentation en espace, indentation de 2

achauffaille replied:
Sachant que ça va bouger ou disparaître je ne touche pas

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
+            $(document).scroll(function() { 
+                scroll_pos = $(this).scrollTop();
+                if(scroll_pos > positionElementInPage4-60) {
+                    $(".pure-menu-link[href='#beta-testeur']").css('background-color', '#FFFFFF');
```
centraliser dans une fonction avec en param : bg_color, color, src_logo, hash

achauffaille replied:
Yé né omprend pas comment faire

existe-deja replied:
ok, je m'en occupe.

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
+                if(scroll_pos > positionElementInPage4-60) {
+                    $(".pure-menu-link[href='#beta-testeur']").css('background-color', '#FFFFFF');
+                    $(".pure-menu-link[href='#beta-testeur']").css('color', '#86c56c');
+                    $(".pure-menu-link[href='#beta-testeur']").css('transition', 'all 0.5s ease');
```
à ajouter dans style.css directement

achauffaille replied:
Je l'aurais bien fait mais je ne sait pas comment ajouter une classe, remove les autre qui étaient attribué à cet élement tout ça tout ça

existe-deja replied:
ok, on en parlera en vrai que tu m'explique le process.

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
+                    $("#press").css('color', '#3381f6');
+                    window.location.hash = 'rejoignez-les-evenements-de-vos-amis';                  
+                }
+                else if(scroll_pos > positionElementInPage2-60) {
```
à quoi correspont - 60 ?

achauffaille replied:
Le -60 permet de modifier le logo un peu en-dessous de la div pour que ce soit plus fluide. Sinon il change trop tôt.

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +31,84 @@ $(document).ready(function(){
+            });
+
+
+/* ANIMATION POPIN VIDEO */
```
à ajouter dans le callback ready

achauffaille replied:
Ok réglé

existe-deja commented on ui/layout.htm:
```diff
@@ -13,6 +24,19 @@
+      ga('send', 'pageview');
+    </script>
+    <script>
+      window.sr = new scrollReveal();
```
déplacer dans main.js

achauffaille replied:
Ok fait

achauffaille replied:
Par contre je sais pas si tu voulais que je mette window.sr = new scrollReveal(); aussi dans le main mais ça empêche le fonctionnement des anims donc je l'ai laissé là

achauffaille committed:
Modifications pour merle

existe-deja commented on ui/js/main.js:
```diff
@@ -31,4 +43,96 @@ $(document).ready(function(){
+
+
+/* FUNCTION REFRESH */
+/*$(document).ready(function(){      
```
à supprimer du coup.

existe-deja committed:
PR function updateStyleMenu

achauffaille committed:
Fix JS

achauffaille committed:
Fix bugs + pop-in video + animation JS + CSS

achauffaille committed:
Fix popin + responsive pop-in

Reste à gérer le scroll top auto quand je quitte la pop-in depuis la
strate du bas. Je n’y arrive pas.

achauffaille committed:
Fix bug JS + HTML

existe-deja committed:
PR clean js

existe-deja committed:
PR pull master
